src/Integration/Binding/CSharpVBProjectBinder.cs

```diff
@@ -39,19 +40,28 @@ internal class CSharpVBProjectBinder : IProjectBinder
         private readonly ICSharpVBAdditionalFileReferenceChecker additionalFileReferenceChecker;
         private readonly IRuleSetSerializer ruleSetSerializer;
         private readonly CreateBindingOperationFunc createBindingOperationFunc;
+        private readonly ILogger logger;
 
         public CSharpVBProjectBinder(IServiceProvider serviceProvider, IFileSystem fileSystem, ILogger logger)
-            : this(serviceProvider, fileSystem, new RuleSetReferenceChecker(serviceProvider, logger), new CSharpVBAdditionalFileReferenceChecker(serviceProvider), GetCreateBindingOperationFunc(serviceProvider, logger))
+            : this(serviceProvider, fileSystem, logger,
```

Comment:
Added the logger as an externally-provided (i.e. non-test) parameter.
Moved the creation of the injectable parameters onto separate lines for readability.